contrib/remote_hsmd/hsmd.c

```diff
@@ -444,41 +515,45 @@ static struct io_plan *init_hsm(struct io_conn *conn,
 	 * will use */
 	c->chainparams = chainparams;
 
-	/* To support integration tests we honor any seed provided
-	 * in the hsm_secret file (testnet only). Otherwise we
-	 * generate a random seed.
-	 */
-	if (!read_test_seed(&hsm_secret)) {
-		randombytes_buf(&hsm_secret, sizeof(hsm_secret));
-	}
-
 	/* Is this a warm start (restart) or a cold start (first time)? */
-	coldstart = access("WARM", F_OK) == -1;
-
-	proxy_stat rv = proxy_init_hsm(&bip32_key_version, chainparams,
-				       coldstart, &hsm_secret,
-				       &node_id, &pubstuff.bip32);
-	if (PROXY_PERMANENT(rv)) {
-		status_failed(STATUS_FAIL_INTERNAL_ERROR,
-		              "proxy_%s failed: %s", __FUNCTION__,
-			      proxy_last_message());
-	}
-	else if (!PROXY_SUCCESS(rv)) {
-		status_unusual("proxy_%s failed: %s", __FUNCTION__,
-			       proxy_last_message());
-		return bad_req_fmt(conn, c, msg_in,
-				   "proxy_%s error: %s", __FUNCTION__,
-				   proxy_last_message());
-	}
+	if (restore_pubstuff(&node_id, &pubstuff.bip32, &bolt12)) {
+		// This is a warm start.
+		proxy_set_node_id(&node_id);
+	} else {
+		// This is a cold start, initialize the remote signer.
+
+		/* To support integration tests we honor any seed provided
+		 * in the hsm_secret file (testnet only). Otherwise we
```

Comment:
So that's still not production worthy, since we want to generate on the signer side.  Is there an impediment for letting the signer generate instead?

Reply:
So this hack allows the integration test suite to use known seeds (and derived things).   We need a way to know if we are running integration tests ...

Reply:
If the integration test doesn't set an env var already, could we change it to set one?

Reply:
But wait.  Below, it tries to read a test seed.  I assume that's written by the integration test?  So in the `if` statement below, if there's no test seed, then shouldn't we get the signer to generate a seed, rather than calling `randombytes_buf`?

Reply:
I'll try letting the remote signer generate in the case where the `hsm_secret` is not being set by the integration tests ...